eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version0/ChannelTypes0.scala

```diff
@@ -0,0 +1,94 @@
+  // race against our htlc-penalty tx, but if that happens a `WatchSpent` will be triggered and we will claim it correctly).
+  // When migrating from these previous types, we can safely set dummy values in the following fields:
+  //  - we only use the `tx` field of `TransactionWithInputInfo` -> no need to completely fill the `InputInfo`
+  //  - `irrevocablySpent` now contains the whole transaction (previously only the txid): we can easily set these when
+  //    one of *our* transactions confirmed, but not when a *remote* transaction confirms. This can only happen for HTLC
+  //    outputs and in these cases we simply remove the entry in `irrevocablySpent`: the channel will set a `WatchSpent`
+  //    which will immediately be triggered and that will let us store the information in `irrevocablySpent`.
+  //  - the `htlcId` in htlc txs is used to detect timed out htlcs and relay them upstream, but it can be safely set to
+  //    0 because the `timedOutHtlcs` in `Helpers.scala` explicitly handle the case where this information is unavailable.
+
+  private def getPartialInputInfo(parentTx: Transaction, childTx: Transaction): InputInfo = {
+    // When using the default commitment format, spending txs have a single input. These txs are fully signed and never
+    // modified: we don't use the InputInfo in closing business logic, so we don't need to fill everything.
+    val input = childTx.txIn.head.outPoint
+    InputInfo(input, parentTx.txOut(input.index.toInt), Nil)
+  }
+
+  case class LocalCommitPublished(commitTx: Transaction, claimMainDelayedOutputTx: Option[Transaction], htlcSuccessTxs: List[Transaction], htlcTimeoutTxs: List[Transaction], claimHtlcDelayedTxs: List[Transaction], irrevocablySpent: Map[OutPoint, ByteVector32]) {
+    def migrate(): channel.LocalCommitPublished = {
+      val htlcTxs = htlcSuccessTxs ++ htlcTimeoutTxs
+      val knownTxs: Map[ByteVector32, Transaction] = (commitTx :: claimMainDelayedOutputTx.toList ::: htlcTxs ::: claimHtlcDelayedTxs).map(tx => tx.txid -> tx).toMap
+      val irrevocablySpentNew = irrevocablySpent.collect { case (outpoint, txid) if knownTxs.contains(txid) => (outpoint, knownTxs(txid)) }
+      val claimMainDelayedOutputTxNew = claimMainDelayedOutputTx.map(tx => ClaimLocalDelayedOutputTx(getPartialInputInfo(commitTx, tx), tx))
+      val htlcSuccessTxsNew = htlcSuccessTxs.map(tx => HtlcSuccessTx(getPartialInputInfo(commitTx, tx), tx, ByteVector32.Zeroes, 0))
+      val htlcTimeoutTxsNew = htlcTimeoutTxs.map(tx => HtlcTimeoutTx(getPartialInputInfo(commitTx, tx), tx, 0))
+      val htlcTxsNew = (htlcSuccessTxsNew ++ htlcTimeoutTxsNew).map(tx => tx.input.outPoint -> Some(tx)).toMap
+      val claimHtlcDelayedTxsNew = claimHtlcDelayedTxs.flatMap(tx => htlcTxs.find(_.txid == tx.txIn.head.outPoint.txid).map(htlcTx => ClaimLocalDelayedOutputTx(getPartialInputInfo(htlcTx, tx), tx)))
```

Comment:
Same here, I would be more aggressive to make migration errors obvious.

Reply:
For this one we should indeed always have the corresponding htlc tx, we can fail if it's not the case.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1728/commits/67616509ca1624defda5b9cf8c252cef23cf7857